src/main/java/com/minecolonies/network/messages/BuildToolPlaceMessage.java

```diff
@@ -140,34 +134,59 @@ public IMessage onMessage(@NotNull BuildToolPlaceMessage message, @NotNull Messa
      */
     private static void handleHut(@NotNull World world, @NotNull EntityPlayer player, String hut, String style, int rotation, @NotNull BlockPos buildPos)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "handleHut" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)